core/inputs/input.ts

```diff
@@ -298,6 +293,10 @@ export class Input {
       this.connection.dispose();
     }
   }
+
+  protected makeConnection(type: number): Connection {
```

Comment:
Consider adding tsdoc for this method to explain to implementers of custom inputs when they should call this method.